components/intake.py

```diff
@@ -18,16 +20,19 @@ def __init__(self):
             ids.PcmChannels.Intake.intake_piston_reverse,
         )
 
-    def execute(self):
+    def deploy(self) -> None:
+        self.deployed = True
+
+    def retract(self) -> None:
+        self.deployed = False
+
+    def execute(self) -> None:
         if self.deployed:
             self.piston.set(DoubleSolenoid.Value.kForward)
             self.motor.set(self.intake_speed)
         else:
             self.piston.set(DoubleSolenoid.Value.kReverse)
             self.motor.set(0.0)
 
-    def deploy(self):
-        self.deployed = True
-
-    def retract(self):
-        self.deployed = False
+    def get_break_beam_sensor_state(self) -> bool:
+        return self.break_beam.get()
```

Comment:
Please test the break beam sensor on a real control system before merging this.

The roboRIO has an internal pull-up resistor on the DIO pins, so the switch being tripped will typically (i.e. unless the switch being unplugged should behave the same way as tripping the switch) connect signal to ground, pulling the pin low.

Reply:
Ok, it now reverses on return.